### ISSUE TITLE
Auto update option for monthly maps

### ISSUE THREAD

**Issue by soredake** [osmandapp/OsmAnd#14395]
# 🚀 feature request

### Description

It would be nice if there is auto update option for monthly maps.

### Describe the solution you'd like

Have an option to auto update monthly maps, with monthly/bimonthly/etc interval.

### Describe alternatives you've considered

Updating it manually.

**Comment by tradavyvy** [comment]
If this option is added, make it so that you can choose witch map. So osmand does not automaticaly update all the onstalled maps.
I keep most usaaps on my phone for quick refference, but i only travel there once a year. No use for me to automaticly update them. But for my netherlands map i could see it being usefull.

**Comment by hypervtechnics** [comment]
This can be done similar to the way Nextcloud Android Client allows to configure Auto-Upload functionality. Meaning the maps will only be updated when the phone is charging, above a certain battery percentage and connected to a Wi-Fi network. If the downloads are interruptable the could be continued the next time the conditions are met. Something like a "we take care of the maps being up to date silently but not immediately".

**Comment by sonora** [comment]
Maps are huge downloads while the meaningfulness of actual deltas from month to month is actually tiny for the general public, often of interest only for active mappers or individual use cases 

But these facts are not obvious to the common user. Hence for the sake of sustainability (energy/data/bandwidth conservation both on the user and server side), i.e. to avoid huge regular periodic downloads just by ignorance or just for fun, I vote for not implrementing an auto-update feature. Mainly on the grounds that  who knows they need an update for any purpose can IMHO be expected to trigger one manually.

**Comment by hypervtechnics** [comment]
I agree on the sustainability part. Maintaing deltas (upgrading up to 6 past full monthly versions to the current one) is one way to tackle this and to reduce bandwidth. You may also ask the user whether auto update is desired. The user may also select an interval in which updates are needed (up to 6 could be fast due to delta others slower due to full download).

**Comment by hypervtechnics** [comment]
Also on the sustainability topic: I think it would be more efficient on storage and bandwidth capabilities to not include the actual map file in a backup by default. Instead just the selection that the user has downloaded this map would be easier. But that is another issue I guess.

**Comment by luroc** [comment]
Who ever was trapped in an area where the streets have been converted to one-way-streets since the last time you remembered to update your map knows how frustrating this can be.

The moment when you need the map is the worst time to be forced to download a huge file. Especially in an area with low signal coverage.

That's why I really whish for an auto update function that I can set and forget. Also I generally don't use wifi and fulfill all my day-to-day internet needs via mobile data, so I'd strongly disagree to make updates dependant on a wifi connection.

**Comment by jm355** [comment]
It could be implemented in such a way that the app will wait a random number of days after an update is available (between 0 and 31) before attempting to auto-update when the phone is charging on wifi. And if the user manually updates before then, then that's the same as it is now. That would avoid periodic spikes, while ensuring nobody is ever more than a month out of date
